src/infinite/index.ts

```diff
@@ -187,7 +189,12 @@ export const infinite = (<Data, Error>(useSWRNext: SWRHook) =>
             (cacheData &&
               !isUndefined(cacheData[i]) &&
               !config.compare(cacheData[i], pageData))
-          if (fn && shouldFetchPage) {
+          if (
+            fn &&
+            (typeof shouldRevalidatePage === 'function'
```

Comment:
should this `shouldRevalidatePage()` result combined with `shouldFetchPage`? looks like if `shouldFetchPage` is false but `shouldRevalidatePage()` returns true we'll still revalidate

Reply:
@huozhi 
Thank you for your feedback!
That's a good point👍 I think that's a technical decision.

The current implementation uses the result of the `revalidate` function and ignores other conditions if it's defined. This is based on the current `revalidate` option, which means the current `useSWRInfinite` with `revalidate: false` never revalidate.

Do you think we should revalidate in some scenarios even if the `revalidate` function returns `false`?

```js
const shouldFetchPage =
  shouldRevalidatePage(pageData, pageArg) || // if false, the following conditions are tested
  revalidateAll ||
  forceRevalidateAll ||
  isUndefined(pageData) ||
  (revalidateFirstPage && !i && !isUndefined(cacheData)) ||
  shouldRevalidateOnMount ||
  (cacheData &&
    !isUndefined(cacheData[i]) &&
    !config.compare(cacheData[i], pageData)
```

If we combine `shouldRevalidatePage` with `shouldFetchPage`, this makes it difficult to disable revalidation for specific pages because revalidation might occur even if `shouldRevalidatePage` returns false.